xbmc/addons/AddonDatabase.cpp

```diff
@@ -823,3 +830,48 @@ bool CAddonDatabase::RemovePackage(const std::string& packageFileName)
   return ExecuteQuery(sql);
 }
 
+bool CAddonDatabase::AddSystemAddon(const std::string &addonID)
+{
+  try
+  {
+    if (NULL == m_pDB.get())
+      return false;
+    if (NULL == m_pDS.get())
+      return false;
+
+    if (!IsSystemAddonKnown(addonID)) // Enabled
+    {
+      std::string sql = PrepareSQL("insert into system(id, addonID) values(NULL, '%s')", addonID.c_str());
+      m_pDS->exec(sql);
+      return true;
+    }
+    return false; // already registered or failed query
+  }
+  catch (...)
+  {
+    CLog::Log(LOGERROR, "%s failed on addon '%s'", __FUNCTION__, addonID.c_str());
+  }
+  return false;
+}
+
+bool CAddonDatabase::IsSystemAddonKnown(const std::string &addonID)
+{
```

Comment:
We are at the database class and I think it's clear enough if we use `IsSystemAddon`, which indicates that it will lookup if the addon is a system addon within our addon database.